File: comandor/src/main/java/com/komandor/komandor/data/database/KomandorDatabase.java
package com.komandor.komandor.data.database;

import com.komandor.komandor.data.database.certificates.Certificate;
import com.komandor.komandor.data.database.certificates.CertificateDao;
import com.komandor.komandor.data.database.chats.Chat;
import com.komandor.komandor.data.database.chats.ChatDao;
import com.komandor.komandor.data.database.contacts.Contact;
import com.komandor.komandor.data.database.contacts.ContactDao;
import com.komandor.komandor.data.database.contacts.ContactsCertificatesDao;
import com.komandor.komandor.data.database.files.File;
import com.komandor.komandor.data.database.files.FileDao;
import com.komandor.komandor.data.database.messages.Message;
import com.komandor.komandor.data.database.messages.MessageDao;
import com.komandor.komandor.data.database.users.User;
import com.komandor.komandor.data.database.users.UserDao;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SimpleSQLiteQuery;

@Database(entities = {User.class, Certificate.class, Contact.class, Chat.class, Message.class, File.class}, version = 2, exportSchema = false)
public abstract class KomandorDatabase extends RoomDatabase {

    public abstract UserDao userDao();

    public abstract CertificateDao certificateDao();

    public abstract ContactDao contactDao();

    public abstract ContactsCertificatesDao contactsCertificatesDao();

    public abstract ChatDao chatDao();

    public abstract MessageDao messageDao();

    public abstract FileDao fileDao();

    public void clearDB() {
        SimpleSQLiteQuery simpleSQLiteQuery = new SimpleSQLiteQuery("DELETE FROM sqlite_sequence");

        beginTransaction();
        clearAllTables();
        query(simpleSQLiteQuery);
        setTransactionSuccessful();
        endTransaction();
    }

}

File: comandor/src/main/java/com/komandor/komandor/data/database/KomandorDatabase.kt
package com.komandor.komandor.data.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import androidx.room.migration.AutoMigrationSpec
import androidx.room.migration.Migration
import androidx.sqlite.db.SupportSQLiteDatabase
import com.komandor.komandor.data.database.KomandorDatabase.Companion.DB_VERSION
import com.komandor.komandor.data.database.model.Chat
import com.komandor.komandor.data.database.dao.ChatDao
import com.komandor.komandor.data.database.model.LocalCertificate
import com.komandor.komandor.data.database.dao.LocalCertificateDao
import com.komandor.komandor.data.database.model.Message
import com.komandor.komandor.data.database.dao.MessageDao
import com.komandor.komandor.data.database.model.User
import com.komandor.komandor.data.database.dao.UserDao
import dagger.hilt.android.qualifiers.ApplicationContext

@Database(
    entities = [
        User::class,
        LocalCertificate::class,
        //Certificate::class,
        //Contact::class,
        Chat::class,
        Message::class,
        //File::class
    ],
    version = DB_VERSION,
    /*
    autoMigrations = [
        AutoMigration (
            from = 4,
            to = 5,
            spec = AppDatabase.MyAutoMigration::class
        )],

     */
    exportSchema = false
)

@TypeConverters(DateTypeConverter::class)
abstract class KomandorDatabase : RoomDatabase() {
    class MyAutoMigration : AutoMigrationSpec

    companion object {
        const val DB_VERSION = 17
        const val NAME = "app.db"
    }

    abstract fun userDao(): UserDao
    abstract fun localCertificateDao(): LocalCertificateDao
    abstract fun chatDao(): ChatDao
    abstract fun messageDao(): MessageDao


/*
    abstract fun certificateDao(): CertificateDao

    abstract fun fileDao(): FileDao

 */

}

fun buildAppDatabase(@ApplicationContext appContext: Context): KomandorDatabase {
    val MIGRATION_5_6 = object : Migration(5, 6) {
        override fun migrate(database: SupportSQLiteDatabase) {

        }
    }
    return Room.databaseBuilder(
        appContext,
        KomandorDatabase::class.java,
        KomandorDatabase.NAME
    ).addMigrations(MIGRATION_5_6)
        .fallbackToDestructiveMigration()
        .build()
}